evaluation/swe_bench/run_infer.py

```diff
@@ -0,0 +1,411 @@
+import asyncio
+import json
+import logging
+import multiprocessing as mp
+import os
+import pathlib
+import subprocess
+import time
+from concurrent.futures import ProcessPoolExecutor
+
+import pandas as pd
+import whatthepatch
+from datasets import load_dataset
+from tqdm import tqdm
+
+from evaluation.swe_bench.swe_env_box import SWEBenchSSHBox
+from opendevin.controller.state.state import State
+from opendevin.core.config import args, config, get_llm_config_arg
+from opendevin.core.logger import get_console_handler
+from opendevin.core.logger import opendevin_logger as logger
+from opendevin.core.main import main
+from opendevin.events.action import MessageAction
+from opendevin.events.serialization.event import event_to_dict
+
+
+def cleanup():
+    print('Cleaning up child processes...')
+    for process in mp.active_children():
+        print(f'Terminating child process: {process.name}')
+        process.terminate()
+        process.join()
+
+
+def codeact_user_response(state: State) -> str:
+    msg = (
+        'Please continue working on the task on whatever approach you think is suitable.\n'
+        'If you think you have modified the code in a way that fixes the issue, please run the following command: <execute_bash> exit </execute_bash>.\n'
+        'IMPORTANT: YOU SHOULD NEVER ASK FOR HUMAN HELP OR USE THE INTERNET TO SOLVE THIS TASK.\n'
+    )
```

Comment:
I think we can leave this as-is for now, but this seems like a hyperparameter. We could also change the prompt (e.g. by allowing CodeAct to use the internet to help solve problems), so it seems that maybe this shouldn't be in the core evaluation code, but rather in some separate place that makes it changeable.

Reply:
Agree -- but we probably won't want SWE-Bench to use web at all.. agent *might* directly go to the original github and copy all the answers there ;)

Reply:
Ah, interesting. That's a good point... But demonstrating the ability to effectively use StackOverflow or something could be interesting. It's a hard line to draw.